bitnami/kafka/templates/scripts-configmap.yaml

```
@@ -109,12 +109,28 @@ data:
     export KAFKA_CFG_ADVERTISED_LISTENERS="INTERNAL://${MY_POD_NAME}.{{ $fullname }}-headless.{{ $releaseNamespace }}.svc.{{ $clusterDomain }}:{{ $interBrokerPort }},CLIENT://${MY_POD_NAME}.{{ $fullname }}-headless.{{ $releaseNamespace }}.svc.{{ $clusterDomain }}:{{ $clientPort }},EXTERNAL://${EXTERNAL_ACCESS_IP}:${EXTERNAL_ACCESS_PORT}"
     {{- end }}
     {{- end }}
-
+    
     {{- if (include "kafka.tlsEncryption" .) }}
-    if [[ -f "/certs/kafka.truststore.jks" ]] && [[ -f "/certs/kafka-${ID}.keystore.jks" ]]; then
+    JKS_TRUSTSTORE_PATH="/certs"
+    {{- if .Values.auth.jksTruststoreSecret }}
+    JKS_TRUSTSTORE_PATH="/truststore"
+    {{- end }}
+
+    JKS_TRUSTSTORE="${JKS_TRUSTSTORE_PATH}/kafka.trustore.jks"
+    JKS_KEYSTORE="/certs/kafka-${ID}.keystore.jks"
+    
+    {{- if .Values.auth.jksTruststore }}
+    JKS_TRUSTSTORE="${JKS_TRUSTSTORE_PATH}/{{ .Values.auth.jksTruststore }}"
+    {{- end }}    
+    
+    {{- if .Values.auth.jksKeystoreSAN }}
+    JKS_KEYSTORE="/certs/{{ .Values.auth.jksKeystoreSAN }}"
+    {{- end }}
```

```suggestion
    {{- if and .Values.auth.jksTruststoreSecret .Values.auth.jksTruststore }}
    JKS_TRUSTSTORE="/truststore/{{ .Values.auth.jksTruststore }}"
    {{- else if .Values.auth.jksTruststoreSecret  }}
    JKS_TRUSTSTORE="/truststore/kafka.trustore.jks"
    {{- else if .Values.auth.jksTruststore  }}
    JKS_TRUSTSTORE="/certs/{{ .Values.auth.jksTruststore }}"
    {{- else }}
    JKS_TRUSTSTORE="/certs/kafka.trustore.jks"
    {{- end }}
    
    {{- if .Values.auth.jksKeystoreSAN }}
    JKS_KEYSTORE="/certs/{{ .Values.auth.jksKeystoreSAN }}"
    {{- else }}
    JKS_KEYSTORE="/certs/kafka-${ID}.keystore.jks"
    {{- end }}
```
I like your suggestion. I did a small fix to support all combinations for the JKS_TRUSTSTORE